### ISSUE TITLE
Error after module update v2.2.0->v2.3.1

### ISSUE THREAD

**Issue by bastelfreak** [puppetlabs/influxdb#98]
## Describe the Bug

On my influxdb node I see the following error since the upgrade:

```
Error: {:"Unable to retrieve puppet telegraf token"=>"undefined method `strip' for {}:Hash"}
```

## Expected Behavior

I expected no error/no changes.

## Environment
- module version 2.3.1
- PE 2021 / Puppet agent 7.24
- influxdb::token is set in hiera-eyaml with convert_to to sensitive

## Additional Context

I assume it's coming from https://github.com/puppetlabs/influxdb/blob/5f919de1afaf3d318875f182fa5e00cf10d9d17b/lib/puppet/functions/influxdb/retrieve_token.rb#L51

**Comment by bastelfreak** [comment]
https://github.com/puppetlabs/influxdb/blame/5f919de1afaf3d318875f182fa5e00cf10d9d17b/lib/puppet/functions/influxdb/retrieve_token.rb#L40

is that line actually correct? The client_options are passed in the header?

**Comment by m0dular** [comment]
Fixed in #99
